projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -110,14 +147,43 @@ export default function MediaSection() {
 						<MediaUpload
 							title={ REPLACE_MEDIA_LABEL }
 							onSelect={ onUpdateMedia }
-							allowedTypes={ ALLOWED_MEDIA_TYPES }
+							allowedTypes={ allowedMediaTypes }
 							render={ replaceMediaRender }
 						/>
 						<Button onClick={ onRemoveMedia } variant="link" isDestructive>
 							{ REMOVE_MEDIA_LABEL }
 						</Button>
 					</>
 				) }
+				{ validationError && (
+					<Notice
+						className={ styles.notice }
+						isDismissible={ true }
+						onDismiss={ onDismissClick }
+						status="warning"
+					>
+						{ validationError === FILE_TYPE_ERROR && (
+							<p>
+								{ __( 'The selected media type is not accepted by these platforms.', 'jetpack' ) }
+							</p>
+						) }
+						{ validationError === FILE_SIZE_ERROR && (
+							<p>{ __( 'The selected media size is too big for these platforms.', 'jetpack' ) }</p>
+						) }
```

Comment:
It might be nice to use an object here, something like this:

```jsx
const validationErrorMessages = {
  FILE_TYPE_ERROR: __( 'The selected media type is not accepted by these platforms.', 'jetpack' ),
  FILE_SIZE_ERROR: __( 'The selected media size is too big for these platforms.', 'jetpack' ),
};

(<p>{validationErrorMessages[validationError]}</p>)
```

Reply:
Good idea, I'll change it!